setup_linac.py

```diff
@@ -0,0 +1,169 @@
+        self,
+        cavityNum,
+        rackObject,
+        ssaClass=SSA,
+        stepperClass=StepperTuner,
+        piezoClass=Piezo,
+    ):
+        super().__init__(cavityNum, rackObject)
+        self._signals: Optional[SetupSignals] = None
+
+        self._ssa_cal_requested: bool = False
+        self._auto_tune_requested: bool = False
+        self._cav_char_requested: bool = False
+        self._rf_ramp_requested: bool = False
+
+        self.running: bool = False
+
+    @property
+    def signals(self):
+        return self._signals
+
+    @signals.setter
+    def signals(self, value: SetupSignals):
+        self._signals = value
+
+    @property
+    def status_message(self):
+        # TODO get from status PV when available
+        raise NotImplementedError("Cavity status PV not yet created")
+
+    @status_message.setter
+    def status_message(self, message):
+        # TODO write to status PV when available
+        print(message)
+        if self.signals:
+            self.signals.status.emit(message)
+
+    # TODO use severity PV and status message when created
+    def set_finished_message(self, message):
+        print(message)
+        if self.signals:
+            self.signals.finished.emit(message)
+
+    # TODO use status PV when implemented
+    @property
+    def script_is_running(self) -> bool:
+        return self.running
+
+    # TODO use PV when available
+    @property
+    def ssa_cal_requested(self) -> bool:
+        return self._ssa_cal_requested
+
+    @ssa_cal_requested.setter
+    def ssa_cal_requested(self, value: bool):
+        self._ssa_cal_requested = value
+
+    # TODO use PVs when available
+    @property
+    def auto_tune_requested(self) -> bool:
+        return self._auto_tune_requested
+
+    @auto_tune_requested.setter
+    def auto_tune_requested(self, value: bool):
+        self._auto_tune_requested = value
+
+    @property
+    def cav_char_requested(self) -> bool:
+        return self._cav_char_requested
+
+    @cav_char_requested.setter
+    def cav_char_requested(self, value: bool):
+        self._cav_char_requested = value
+
+    @property
+    def rf_ramp_requested(self) -> bool:
+        return self._rf_ramp_requested
+
+    @rf_ramp_requested.setter
+    def rf_ramp_requested(self, value: bool):
+        self._rf_ramp_requested = value
+
+    def capture_acon(self):
+        self.acon = self.ades
+
+    def shut_down(self):
+        self.running = True
+        self.status_message = f"Shutting {self} down"
+        self.turnOff()
+        self.ssa.turn_off()
+        self.set_finished_message(f"{self} RF and SSA off")
+        self.running = False
+
+    def setup(self):
+        self.running = True
+
+        self.status_message = f"Turning {self} off"
+        self.turnOff()
+
+        self.status_message = f"Turning on {self} SSA if not on already"
+        self.ssa.turn_on()
+
+        self.status_message = f"Resetting {self} interlocks"
+        self.reset_interlocks()
+
```

Comment:
Is this section a candidate for case/match instead?

Reply:
All the ifs? Unless I'm misinterpreting your point, they're not exclusive. Like, sometimes you want all of them to run, sometimes you want a subset to run, so I think figuring out all the permutations would make for _a lot_ of cases right?

Reply:
Yep I meant the ifs. Ok, makes sense if they aren't exclusive.